client/src/pages/admin/CreateProduct.test.js

```diff
@@ -0,0 +1,259 @@
+// CreateProduct.test.js
+import React from "react";
+import { render, screen, fireEvent, waitFor } from "@testing-library/react";
+import { act } from "react-dom/test-utils"
+import userEvent from "@testing-library/user-event";
+import "@testing-library/jest-dom";
+import CreateProduct from "./CreateProduct";
+import axios from "axios";
+import toast from "react-hot-toast";
+
+// TODO: Need to settle the Form Submission test which I AM STUCK AT!!!!
+jest.mock("./../../components/Layout", () => {
+  return ({ children, title }) => (
+    <div data-testid="mock-layout" data-title={title}>
+      {children}
+    </div>
+  );
+});
+
+jest.mock("./../../components/AdminMenu", () => {
+  return () => <div data-testid="mock-adminmenu">Admin Menu</div>;
+});
+
+const mockedNavigate = jest.fn();
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useNavigate: () => mockedNavigate,
+}));
+
+jest.mock("axios");
+jest.mock("react-hot-toast");
+global.URL.createObjectURL = jest.fn(() => 'blob:http://localhost/dummy'); // Mock the damn URL for the picture
+
+
+describe("CreateProduct Component", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
```

Comment:
Maybe can add this in the beforeEach block as the below statement was initialised a few times 
    `axios.get.mockResolvedValueOnce({
      data: { success: true, category: [] },
    });`